src/java.desktop/macosx/classes/sun/lwawt/macosx/CTrayIcon.java

```diff
@@ -173,7 +173,6 @@ public void dispose() {
         }
 
         LWCToolkit.targetDisposedPeer(target, this);
-        target = null;
```

Comment:
to make sure it is not changed in some other place you can merk the target field as final.